qemu/tests/block_boot_multi_disks.py

```diff
@@ -0,0 +1,101 @@
+from virttest.utils_misc import get_linux_drive_path
+from virttest.utils_windows.drive import get_disk_props_by_serial_number
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test booting with multi disks
+
+    1) Boot the vm with multi disk.
+    2) Check BIOS log for SEABios booting.
+    3) Login guest check the disk number.
+    4) Execute IO operation on some disks.
+    """
+
+    def _prepare_images():
+        image_name = params.get("stg_image_name", "images/%s")
+        drive_format = params.get("stg_drive_format", "virtio")
+        image_format = params.get("stg_image_format", "qcow2")
+
+        for idx_ in range(stg_image_num):
+            name = "stg%d" % idx_
+            params["images"] = params["images"] + " " + name
+            params["image_name_%s" % name] = image_name % name
+            params["image_size_%s" % name] = image_size
+            params["image_format_%s" % name] = image_format
+            params["drive_format_%s" % name] = drive_format
+            params["boot_drive_%s" % name] = "yes"
+            params["blk_extra_params_%s" % name] = "serial=%s" % name
+
+    def _get_window_disk_index_by_serial(serial):
+        idx_info = get_disk_props_by_serial_number(session, serial, ["Index"])
+        if idx_info:
+            return idx_info["Index"]
+        test.fail("Not find expected disk %s" % serial)
+
+    def _check_disk_in_guest(img):
+        logger.debug("Check disk %s in guest" % img)
+        if os_type == 'windows':
+
+            cmd = utils_misc.set_winutils_letter(session, guest_cmd)
+            disk = _get_window_disk_index_by_serial(img)
+            utils_disk.update_windows_disk_attributes(session, disk)
+            logger.info("Formatting disk:%s", disk)
+            driver = \
+                utils_disk.configure_empty_disk(session, disk, image_size,
+                                                os_type)[0]
+            output_path = driver + ":\\test.dat"
+            cmd = cmd.format(output_path)
+        else:
+            output_path = get_linux_drive_path(session, img)
+            cmd = guest_cmd.format(output_path)
+
+        logger.debug(cmd)
+        session.cmd(cmd)
+
+    logger = test.log
+    stg_image_num = params.get_numeric('stg_image_num')
+    os_type = params["os_type"]
+    image_size = params.get("stg_image_size", "512M")
+    guest_cmd = params["guest_cmd"]
+
+    logger.info('Prepare images ...%s', params["stg_image_num"])
+    _prepare_images()
+    logger.info('Booting vm...')
+    params['start_vm'] = 'yes'
+    vm = env.get_vm(params['main_vm'])
+    env_process.process(test, params, env, env_process.preprocess_image,
+                        env_process.preprocess_vm)
+    timeout = int(params.get("login_timeout", 360))
+
+    logger.debug("Login in guest...")
+    session = vm.wait_for_login(timeout=timeout)
+
+    check_message = params.get("check_message")
+    if check_message:
```

Comment:
For aarch64 we do not have BIOS log, so we need to distinguish this parameter in cfg.